Code/Framework/AzToolsFramework/AzToolsFramework/AssetBrowser/AssetBrowserThumbnailViewProxyModel.cpp

```diff
@@ -115,5 +120,109 @@ namespace AzToolsFramework
                 endResetModel();
             }
         }
+
+        Qt::DropActions AssetBrowserThumbnailViewProxyModel::supportedDropActions() const
+        {
+            return Qt::CopyAction | Qt::MoveAction;
+        }
+
+        Qt::ItemFlags AssetBrowserThumbnailViewProxyModel::flags(const QModelIndex& index) const
+        {
+            Qt::ItemFlags defaultFlags = QAbstractItemModel::flags(index) | Qt::ItemIsEditable;
+
+            if (index.isValid())
+            {
+                // We can only drop items onto folders so set flags accordingly
+                const AssetBrowserEntry* item = mapToSource(index).data(AssetBrowserModel::Roles::EntryRole).value<const AssetBrowserEntry*>();
+                if (item)
+                {
+                    if (item->RTTI_IsTypeOf(ProductAssetBrowserEntry::RTTI_Type()) || item->RTTI_IsTypeOf(SourceAssetBrowserEntry::RTTI_Type()))
+                    {
+                        return Qt::ItemIsDragEnabled | defaultFlags;
+                    }
+                    if (item->RTTI_IsTypeOf(FolderAssetBrowserEntry::RTTI_Type()))
```

Comment:
Just use `item->GetEntryType()` here instead of performing these RTTI casts.

Reply:
Done